client/setec/store.go

```diff
@@ -349,22 +377,55 @@ func StaticFile(path string) (Secret, error) {
 	return func() []byte { return bs }, nil
 }
 
+// hasExpired reports whether cs is an undeclared secret whose last access time
+// was longer ago than the expiry window.
+func (s *Store) hasExpired(cs *cachedSecret) bool {
+	if cs.Declared {
+		return false // declared secrets do not expire
+	} else if s.expiryAge <= 0 {
+		return false // no expiry age is defined
+	}
+	age := time.Unix(s.timeNow(), 0).UTC().Sub(cs.lastAccessTime())
```

Comment:
this to me looks like s.timeNow should return the (pluggable) equivalent of time.Now().UTC(), and let the caller decide if they need the unix timestamp. It looks jarring to have a time function that needs plumbing through the time.Unix constructor to use for basic time arithmetic.

Reply:
As noted elsethread, I reworked this as you suggested to report a time.Time, and put in the conversions explicitly where used. (Only one place actually cares about UTC, so that was indeed simpler as you suggested)